WalletWasabi.Gui/Controls/WalletExplorer/SendControlViewModel.cs

```diff
@@ -871,63 +872,64 @@ private bool TryParseUserFee(out decimal userFee)
 				&& userFee > 0;
 		}
 
-		public ErrorDescriptors ValidateUserFeeText()
+		private void ValidateUserFeeText(IValidationErrors errors)
 		{
-			return TryParseUserFee(out _)
-				? ErrorDescriptors.Empty
-				: new ErrorDescriptors(new ErrorDescriptor(ErrorSeverity.Error, "Invalid fee."));
+			if (!TryParseUserFee(out _))
+			{
+				errors.Add(ErrorSeverity.Error, "Invalid fee.");
+			}
 		}
 
-		public ErrorDescriptors ValidatePassword() => PasswordHelper.ValidatePassword(Password);
+		private void ValidatePassword(IValidationErrors errors) => PasswordHelper.ValidatePassword(errors, Password);
 
-		public ErrorDescriptors ValidateActiveAddress()
+		private void ValidateAddress(IValidationErrors errors)
 		{
 			if (string.IsNullOrWhiteSpace(Address))
 			{
-				return ErrorDescriptors.Empty;
+				return;
 			}
 
 			if (Address == CustomChangeAddress)
 			{
-				return new ErrorDescriptors(new ErrorDescriptor(ErrorSeverity.Error, "The active address and the change address cannot be the same."));
+				errors.Add(ErrorSeverity.Error, "The active address and the change address cannot be the same.");
```

Comment:
Return after the add.

Reply:
why? it is possible that the addresses can be the same and invalid?

Reply:
ACK